File: src/com/eagle/programmar/CSharp/CSharp_Using.java
// Copyright Eagle Legacy Modernization LLC, 2010-date
// Original author: Steven A. O'Hara, Jul 2, 2024

package com.eagle.programmar.CSharp;

import com.eagle.core.EagleInterpreter;
import com.eagle.core.EagleRunnable;
import com.eagle.programmar.CSharp.Terminals.CSharp_Identifier;
import com.eagle.programmar.CSharp.Terminals.CSharp_Keyword;
import com.eagle.programmar.CSharp.Terminals.CSharp_Punctuation;
import com.eagle.tokens.SeparatedList;
import com.eagle.tokens.TokenList;
import com.eagle.tokens.TokenSequence;
import com.eagle.tokens.punctuation.PunctuationEquals;
import com.eagle.tokens.punctuation.PunctuationPeriod;
import com.eagle.tokens.punctuation.PunctuationSemicolon;

public class CSharp_Using extends TokenSequence implements EagleRunnable
{
	public @S(10) CSharp_Keyword USING = new CSharp_Keyword("using");
	public @S(20) @OPT CSharp_Keyword STATIC = new CSharp_Keyword("static");
	public @S(30) CSharp_Identifier id;
	public @S(40) @OPT TokenList<CSharp_MoreUsing> moreIds;
	public @S(50) @OPT CSharp_UsingEquals alternateName;
	public @S(60) @NOSPACE PunctuationSemicolon semicolon;

	public static class CSharp_MoreUsing extends TokenSequence
	{
		public @S(10) @NOSPACE PunctuationPeriod dot;
		public @S(20) @NOSPACE CSharp_Identifier id;
	}

	public static class CSharp_UsingEquals extends TokenSequence
	{
		public @S(10) PunctuationEquals equals;
		public @S(20) @OPT CSharp_UsingGlobal global;
		public @S(30) SeparatedList<CSharp_Identifier, PunctuationPeriod> id;

		public static class CSharp_UsingGlobal extends TokenSequence
		{
			public @S(10) CSharp_Keyword GLOBAL = new CSharp_Keyword("global");
			public @S(20) CSharp_Punctuation colonColon = new CSharp_Punctuation("::");
		}
	}

	@Override
	public void interpret(EagleInterpreter interpreter)
	{
		// Nothing to do here
	}
}

File: csrc/com/eagle/programmar/CSharp/CSharp_Using.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// Purchase a Premium Edition license at:
// https://www.tangiblesoftwaresolutions.com/order/order-java-to-csharp.html
// ====================================================================================================

// Copyright Eagle Legacy Modernization LLC, 2010-date
// Original author: Steven A. O'Hara, Jul 2, 2024

namespace com.eagle.programmar.CSharp
{
	using EagleInterpreter = com.eagle.interpret.EagleInterpreter;
	using EagleRunnable = com.eagle.interpret.EagleRunnable;
	using CSharp_Identifier = com.eagle.programmar.CSharp.Terminals.CSharp_Identifier;
	using CSharp_Keyword = com.eagle.programmar.CSharp.Terminals.CSharp_Keyword;
	using CSharp_Punctuation = com.eagle.programmar.CSharp.Terminals.CSharp_Punctuation;
	using SeparatedList = com.eagle.tokens.SeparatedList;
	using TokenList = com.eagle.tokens.TokenList;
	using TokenSequence = com.eagle.tokens.TokenSequence;
	using PunctuationEquals = com.eagle.tokens.punctuation.PunctuationEquals;
	using PunctuationPeriod = com.eagle.tokens.punctuation.PunctuationPeriod;
	using PunctuationSemicolon = com.eagle.tokens.punctuation.PunctuationSemicolon;

	public class CSharp_Using : TokenSequence, EagleRunnable
	{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) com.eagle.programmar.CSharp.Terminals.CSharp_Keyword USING = new com.eagle.programmar.CSharp.Terminals.CSharp_Keyword("using");
		public CSharp_Keyword USING = new CSharp_Keyword("using");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) @OPT CSharp_Keyword STATIC = new com.eagle.programmar.CSharp.Terminals.CSharp_Keyword("static");
		public  OPT;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(30) com.eagle.programmar.CSharp.Terminals.CSharp_Identifier id;
		public CSharp_Identifier id;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(40) @OPT TokenList<CSharp_MoreUsing> moreIds;
		public  OPT;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(50) @OPT CSharp_UsingEquals alternateName;
		public  OPT;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(60) @NOSPACE PunctuationSemicolon semicolon;
		public  NOSPACE;

		public class CSharp_MoreUsing : TokenSequence
		{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) @NOSPACE PunctuationPeriod dot;
			public  NOSPACE;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) @NOSPACE CSharp_Identifier id;
			public  NOSPACE;
		}

		public class CSharp_UsingEquals : TokenSequence
		{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) com.eagle.tokens.punctuation.PunctuationEquals equals;
			public PunctuationEquals equals;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) @OPT CSharp_UsingGlobal global;
			public  OPT;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(30) com.eagle.tokens.SeparatedList<com.eagle.programmar.CSharp.Terminals.CSharp_Identifier, com.eagle.tokens.punctuation.PunctuationPeriod> id;
			public SeparatedList<CSharp_Identifier, PunctuationPeriod> id;

			public class CSharp_UsingGlobal : TokenSequence
			{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) com.eagle.programmar.CSharp.Terminals.CSharp_Keyword GLOBAL = new com.eagle.programmar.CSharp.Terminals.CSharp_Keyword("global");
				public CSharp_Keyword GLOBAL = new CSharp_Keyword("global");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) com.eagle.programmar.CSharp.Terminals.CSharp_Punctuation colonColon = new com.eagle.programmar.CSharp.Terminals.CSharp_Punctuation("::");
				public CSharp_Punctuation colonColon = new CSharp_Punctuation("::");
			}
		}

		public override void interpret(EagleInterpreter interpreter)
		{
			// Nothing to do here
		}
	}

}
